File: Modules/TaskFlow/WorkspaceMember/WorkspaceMemberColumns.cs
namespace TaskFlowSerenity.TaskFlow.Columns;

[ColumnsScript("TaskFlow.WorkspaceMember")]
[BasedOnRow(typeof(WorkspaceMemberRow), CheckNames = true)]
public class WorkspaceMemberColumns
{
    [EditLink]
    public string WorkspaceName { get; set; }

    public string UserDisplayName { get; set; }

    [EditLink]
    public string Role { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: Modules/ServerTypes/TaskFlow/WorkspaceMemberColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WorkspaceMemberRow } from "./WorkspaceMemberRow";

export interface WorkspaceMemberColumns {
    WorkspaceName: Column<WorkspaceMemberRow>;
    UserDisplayName: Column<WorkspaceMemberRow>;
    Role: Column<WorkspaceMemberRow>;
    CreatedAt: Column<WorkspaceMemberRow>;
    UpdatedAt: Column<WorkspaceMemberRow>;
}

export class WorkspaceMemberColumns extends ColumnsBase<WorkspaceMemberRow> {
    static readonly columnsKey = 'TaskFlow.WorkspaceMember';
    static readonly Fields = fieldsProxy<WorkspaceMemberColumns>();
}